internal/slicer/slicer.go

```diff
@@ -342,3 +347,17 @@ func contains(l []string, s string) bool {
 	}
 	return false
 }
+
+// PackageArchive takes in a package and a map of archives by archive names and
+// returns the appropriate archive for the specified package.
+func PackageArchive(pkg *setup.Package, archives map[string]archive.Archive) (archive.Archive, error) {
+	archiveName := pkg.Archive
+	archive := archives[archiveName]
+	if archive == nil {
+		return nil, fmt.Errorf("archive %q not defined", archiveName)
+	}
+	if !archive.Exists(pkg.Name) {
+		return nil, fmt.Errorf("slice package %q missing from archive", pkg.Name)
+	}
+	return archive, nil
+}
```

Comment:
I think this should be outside. In fact, we are moving archive selection outside so that `cmd_cut` knows exactly which archives are going to be used which involves seeing if the package is actually there. I think `slicer` should no longer be concern about this, this function should not live here.

Reply:
Moving this to the `archive` package instead of moving it to `main` (in `cmd_cut` or similar place). The reason being this will still be used in `slicer_test`.